crates/holochain/src/core/ribosome/host_fn/get_details.rs

```diff
@@ -15,44 +15,58 @@ pub fn get_details<'a>(
     inputs: Vec<GetInput>,
 ) -> Result<Vec<Option<Details>>, WasmError> {
     match HostFnAccess::from(&call_context.host_context()) {
-        HostFnAccess{ read_workspace: Permission::Allow, .. } => {
-            let results: Vec<Result<Option<Details>, _>> = tokio_helper::block_forever_on(async move {
-                join_all(inputs.into_iter().map(|input| {
-                    async {
+        HostFnAccess {
+            read_workspace: Permission::Allow,
+            ..
+        } => {
+            let results: Vec<Result<Option<Details>, _>> =
+                tokio_helper::block_forever_on(async move {
+                    join_all(inputs.into_iter().map(|input| async {
                         let GetInput {
                             any_dht_hash,
                             get_options,
                         } = input;
                         Cascade::from_workspace_network(
                             &call_context.host_context.workspace(),
                             call_context.host_context.network().to_owned(),
-                        ).get_details(any_dht_hash, get_options).await
-                    }
-                })).await
-            });
-            let results: Result<Vec<_>, _> = results.into_iter().map(|result| result.map_err(|cascade_error| WasmError::Host(cascade_error.to_string()))).collect();
+                        )
+                        .get_details(any_dht_hash, get_options)
+                        .await
+                    }))
+                    .await
+                });
+            let results: Result<Vec<_>, _> = results
+                .into_iter()
+                .map(|result| {
+                    result.map_err(|cascade_error| WasmError::Host(cascade_error.to_string()))
+                })
+                .collect();
             Ok(results?)
-        },
-        _ => Err(WasmError::Host(RibosomeError::HostFnPermissions(
-            call_context.zome.zome_name().clone(),
-            call_context.function_name().clone(),
-            "get_details".into()
-        ).to_string()))
+        }
+        _ => Err(WasmError::Host(
+            RibosomeError::HostFnPermissions(
+                call_context.zome.zome_name().clone(),
+                call_context.function_name().clone(),
+                "get_details".into(),
+            )
+            .to_string(),
+        )),
     }
 }
 
 #[cfg(test)]
 #[cfg(feature = "slow_tests")]
 pub mod wasm_test {
-    use crate::fixt::ZomeCallHostAccessFixturator;
-    use ::fixt::prelude::*;
     use hdk::prelude::*;
     use holochain_wasm_test_utils::TestWasm;
+    use crate::core::ribosome::wasm_test::RibosomeTestFixture;
 
     #[tokio::test(flavor = "multi_thread")]
-    async fn ribosome_get_details_test<'a>() {
+    async fn ribosome_get_details_test() {
         observability::test_run().ok();
-        let host_access = fixt!(ZomeCallHostAccess, Predictable);
+        let RibosomeTestFixture {
```

Comment:
So it might be worth changing the timeout in the network config for this test. If it does a lot of gets then it could be very slow if it waits 30 seconds to find out something isn't there for each call. Have you checked how long this takes to run?

Reply:
test core::ribosome::host_fn::get_details::wasm_test::ribosome_get_details_test ... ok <1.070s>

1 second for this one